feat: update for tricrypto crv gauge

0xDosa opened the pull request.

protodev-rage committed:
feat: update for tricrypto crv gauge

134dd3v commented on scripts/check.ts:
```diff
@@ -0,0 +1,208 @@
+import { CurveYieldStrategy__factory, vaults } from '@ragetrade/sdk';
```
would this script be required in repo?

134dd3v commented on test/fixtures/addresses.ts:
```diff
@@ -4,6 +4,8 @@ export default {
   USDC: '0xff970a61a04b1ca14834a43f5de4533ebddb5cc8',
   USDT: '0xfd086bc7cd5c481dcc9c85ebe478a1c0b69fcbb9',
   GAUGE: '0x97E2768e8E73511cA874545DC5Ff8067eB19B787',
```
Should this be old gauge, just for someone without context?

```suggestion
  OLD_GAUGE: '0x97E2768e8E73511cA874545DC5Ff8067eB19B787',
```

protodev-rage committed:
fix: revoke allowance on old gauge

134dd3v reviewed (commented):
LGTM. Just few things as todo:
- We should document this properly/include context for external viewer, maybe in our docs site, this PR or in README.
- add storage layout tests
- move this to a separate repo later (if we prefer this)

0xDosa committed:
ref: changed gauge to old_gauge

protodev-rage committed:
test: remove claimable_reward_write and update tests accordingly

134dd3v committed:
test: fix storage layout tests

Due to some changes in code, ast id is changed however the variables are
still on the same slot and offset. This commit fixes the logic to get
the storage entry.

protodev-rage committed:
test: fix WETH_WHALE address

134dd3v commented on contracts/libraries/Logic.sol:
```diff
@@ -305,4 +306,16 @@ library Logic {
+
+        uint256 bal = oldGauge.balanceOf(address(this));
+
+        IERC20(0x8e0B8c8BB9db49a46697F3a5Bb8A308e744821D2).approve(address(oldGauge), 0);
```
Here I think we should mention which token this is, probably in a comment or just create an intermediate variable variable.
```suggestion
        IERC20 crv3 = IERC20(0x8e0B8c8BB9db49a46697F3a5Bb8A308e744821D2);
        crv3.approve(address(oldGauge), 0);
```

134dd3v commented on contracts/yieldStrategy/CurveYieldStrategy.sol:
```diff
@@ -161,11 +165,15 @@ contract CurveYieldStrategy is EightyTwentyRangeStrategyVault {
     function _harvestFees() internal override {
-        uint256 claimable = gauge.claimable_reward(address(this), address(crvToken)) + crvPendingToSwap;
+        uint256 before = crvToken.balanceOf(address(this));
+        IGaugeFactory(0xabC000d88f23Bb45525E447528DBF656A9D55bf5).mint(address(gauge));
```
Should we move this `IGaugeFactory` to the top as a constant? It should not break storage layout, as well as reduce inline hardcoding.

134dd3v commented on scripts/check.ts:
```diff
@@ -0,0 +1,208 @@
+import { CurveYieldStrategy__factory, vaults } from '@ragetrade/sdk';
```
We should either remove this file if we are using it for informal checking or rename it so that its purpose is clear.

134dd3v replied:
Update: protodev is moving this to a test case.

134dd3v commented on scripts/get-storage-layout.ts:
```diff
@@ -0,0 +1,54 @@
+import fs from 'fs';
```
This file is also to be removed since it's there in helpers.

134dd3v commented on test/CurveYieldStrategy.spec.ts:
```diff
@@ -506,7 +500,7 @@ describe('CurveYieldStrategy', () => {
     });
 
-    it('should deduct rage fee (10% which can be changed)', async () => {
+    it.skip('should deduct rage fee (10% which can be changed)', async () => {
```
Is the `.skip` intended? Since it's skipped during changes on audited code, we should mention the context either around it or in the commit message.

134dd3v replied:
Update: protodev would add a comment for now explaining the reason, and later will update this test case.

134dd3v committed:
ref: review changes

- removed get-storage-layout file from script dir
- add name to contract in Logic file
- move hardcoded addresses at top (except from migrate method, since
it is to be removed)

protodev-rage committed:
test: update impl

protodev-rage committed:
test: update impl

protodev-rage committed:
feat: add recipient for withdrawing fees

protodev-rage committed:
fix: spelling and convert to snake case

protodev-rage committed:
fix: add missing params

protodev-rage committed:
test: deposit fails if midway in migrating

protodev-rage committed:
test: remove .only
